internal/contract/types.go

```diff
@@ -166,3 +198,124 @@ func (i *Duration) Set(obj *unstructured.Unstructured, value metav1.Duration) er
 	}
 	return nil
 }
+
+// MachineAddresses represents an accessor to a []clusterv1.MachineAddress path value.
+type MachineAddresses struct {
```

Comment:
If this type is specific only of infra machine, let's consider to move the implementation in the corresponding file (same for FailureDomains below)